### PR TITLE
bump k8s packages to 1.17.6

### PR DESCRIPTION
**Pull request by dprotaso**
- [x] https://github.com/knative/serving/pull/7997
- [x] https://github.com/knative/eventing/pull/3163
- [x] https://github.com/knative/eventing-contrib/pull/1243
- [x] https://github.com/knative/sample-controller/pull/282
- [x] https://github.com/knative/sample-source/pull/81

### CODE REVIEW

File: hack/update-k8s-deps.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  "k8s.io/apiextensions-apiserver"
+  "k8s.io/apimachinery"
+  "k8s.io/code-generator"
+  "k8s.io/client-go"
+)
+
+
+for dep in "${K8S_DEPS[@]}"
+do
+  go mod edit \
```
Comment (markusthoemmes) [inline hack/update-k8s-deps.sh:35]: Nice trick, I'm stealing this for release automation if you don't mind! :joy:

**Comment by vagababov** [comment]
I guess doesn't matter, since we pinned pkg.

**Comment by dprotaso** [comment]
> I guess doesn't matter, since we pinned pkg.

yeah

**Comment by dprotaso** [comment]
All downstream PRs are passing their required checks

**Comment by dprotaso** [comment]
Curious if this works ;)

/approve
/lgtm

**Comment by dprotaso** [comment]
welp - there you have it

**Comment by vagababov** [comment]
/assign @mattmoor 
/lgtm
/hold
:) There you go

**Comment by vagababov** [comment]
I wonder why does it need `approve`.
/verify-owners

**Comment by vagababov** [comment]
Because it's pkg 🥺

**Comment by dprotaso** [comment]
yeah 
/hold cancel

**Comment by dprotaso** [comment]
thanks @vaikas

**Pull request merged**